src/test/getarg_tests.cpp

```diff
@@ -13,9 +13,18 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/test/unit_test.hpp>
 
-BOOST_FIXTURE_TEST_SUITE(getarg_tests, BasicTestingSetup)
+namespace getarg_tests{
+    class LocalTestingSetup : BasicTestingSetup {
+        protected:
+        void SetupArgs(ArgsManager& m_args, const std::vector<std::pair<std::string, unsigned int>>& args);
```

Comment:
```suggestion
        void SetupArgs(const std::vector<std::pair<std::string, unsigned int>>& args);
```

This should be removed (along with the first commit), because in a member function you don't need to pass in member variables anymore